orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/MonitoredDeployBaseTask.java

```diff
@@ -132,10 +190,60 @@ private TaskResult handleError(
         monitorDefinition,
         e);
 
-    return TaskResult.builder(ExecutionStatus.SUCCEEDED)
-        // TODO(mvulfson)
-        // .context()
-        .build();
+    String userMessage =
+        String.format(
+            "Failed to get a valid response from deployment monitor %s, proceeding anyway because this deployment monitor is configured to not failOnError",
+            monitorDefinition.getName());
+
+    return buildTaskResult(TaskResult.builder(ExecutionStatus.SUCCEEDED), userMessage);
+  }
+
+  private void ensureMonitorDefinition(Stage stage) {
+    if (this.stage == null) {
+      MonitoredDeployStageData context = stage.mapTo(MonitoredDeployStageData.class);
+      this.stage = stage;
+      this.monitorDefinition =
+          deploymentMonitorServiceProvider.getDefinitionById(
+              context.getDeploymentMonitor().getId());
+    }
+  }
+
+  TaskResult buildTaskResult(
+      TaskResult.TaskResultBuilder taskResultBuilder, EvaluateHealthResponse response) {
+    List<StatusReason> statusReasons =
+        Optional.ofNullable(response.getStatusReasons()).orElse(Collections.emptyList());
+
+    String summary = "Deployment monitor requested to: " + response.getNextStep().getDirective();
+    StatusExplanation explanation = new StatusExplanation(summary, statusReasons);
+
+    return taskResultBuilder.context("deploymentMonitorReasons", explanation).build();
+  }
+
+  TaskResult buildTaskResult(TaskResult.TaskResultBuilder taskResultBuilder, String summary) {
+    StatusExplanation explanation = new StatusExplanation(summary);
+
+    return taskResultBuilder.context("deploymentMonitorReasons", explanation).build();
+  }
+
+  String getRetrofitLogMessage(Response response) {
+    if (response == null) {
+      return "<NO RESPONSE>";
+    }
+
+    String body = "";
+    String status = "";
+    String headers = "";
+
+    try {
+      status = String.format("%d (%s)", response.getStatus(), response.getReason());
+      body = CharStreams.toString(new InputStreamReader(response.getBody().in()));
```

Comment:
it seems like we should specify an encoding somewhere in here

Reply:
you think `utf-8`? i don't even know what the default for retrofit (accept header) is... let me dig

Reply:
I'm a bit ignorant about this but I'd say `utf-8` all the things is probably better than relying on some unspecified system default

Reply:
yes, agree, updated.. apparently... google/android JVM defaults to UTF-8 (which of course can be changed). Others seem to have VERY convoluted logic to figure out the default encoding depending on your locale 😨